R/add_keep_population_flag.R

```diff
@@ -0,0 +1,163 @@
+      unique()
+    if (calendar_year %in% year_available) {
+      pop_estimates <- pop_estimates %>%
+        dplyr::filter(year == calendar_year)
+    } else {
+      previous_year <- sort(year_available, decreasing = TRUE)[1]
+      pop_estimates <- pop_estimates %>%
+        dplyr::filter(year = previous_year)
+    }
+
+    pop_estimates <- pop_estimates %>%
+      # Recode gender to make it match source.
+      dplyr::mutate(sex = dplyr::if_else(sex == "M", 1, 2)) %>%
+      dplyr::rename(
+        "age90" = "age90plus",
+        "gender" = "sex"
+      ) %>%
+      tidyr::pivot_longer(
+        names_to = "age",
+        names_prefix = "age",
+        values_to = "population_estimate",
+        cols = "age0":"age90"
+      ) %>%
+      dplyr::mutate(
+        age = as.integer(age),
+        age_group = dplyr::case_when(
+          age >= 0 & age <= 4 ~ "0-4",
+          age >= 5 & age <= 14 ~ "5-14",
+          age >= 15 & age <= 24 ~ "15-24",
+          age >= 25 & age <= 34 ~ "25-34",
+          age >= 35 & age <= 44 ~ "35-44",
+          age >= 45 & age <= 54 ~ "45-54",
+          age >= 55 & age <= 64 ~ "55-64",
+          age >= 65 & age <= 74 ~ "65-74",
+          age >= 75 & age <= 84 ~ "75-84",
+          age >= 85 ~ "85+"
+        )
+      ) %>%
+      dplyr::left_join(
+        get_locality_path() %>%
+          readRDS() %>%
+          dplyr::select("locality" = "hscp_locality", datazone2011),
+        by = "datazone2011"
+      ) %>%
+      dplyr::group_by(locality, age_group, gender) %>%
+      dplyr::summarize(population_estimate = sum(population_estimate)) %>%
+      dplyr::ungroup()
+
+    # Step 2: Work out the current population sizes in the SLF for Locality, AgeGroup, and Gender
+    # Work out the current population sizes in the SLF for Locality AgeGroup and Gender.
+    individual_file <- slfhelper::read_slf_individual(year,
+      columns = c(
+        "chi",
+        "locality",
+        "age",
+        "gender",
+        # "nsu",
+        "death_date"
+      )
+    ) %>%
+      dplyr::mutate(nsu = 0L) # delete this before merge
+
+    individual_file_1 <- individual_file %>%
+      dplyr::mutate(
+        age = as.integer(age),
+        age_group = dplyr::case_when(
+          age >= 0 & age <= 4 ~ "0-4",
+          age >= 5 & age <= 14 ~ "5-14",
+          age >= 15 & age <= 24 ~ "15-24",
+          age >= 25 & age <= 34 ~ "25-34",
+          age >= 35 & age <= 44 ~ "35-44",
+          age >= 45 & age <= 54 ~ "45-54",
+          age >= 55 & age <= 64 ~ "55-64",
+          age >= 65 & age <= 74 ~ "65-74",
+          age >= 75 & age <= 84 ~ "75-84",
+          age >= 85 ~ "85+"
+        )
+      )
+
+    set.seed(100)
+    mid_year <- lubridate::dmy(stringr::str_glue("30-06-{calendar_year}"))
+    ## issues with age being negative
+    # If they don't have a locality, they're no good as we won't have an estimate to match them against.
+    # Same for age and gender.
+    nsu_keep_lookup <- individual_file_1 %>%
```

Comment:
I got a bit lost from here onwards as the nsu variable only had 0 for each row. We can have chat about this on tuesday